service/api.go

```diff
@@ -616,19 +615,13 @@ func (as *APIServer) listVariants(w http.ResponseWriter, r *http.Request) {
 func (as *APIServer) validateProjectConfig(w http.ResponseWriter, r *http.Request) {
 	body := utility.NewRequestReader(r)
 	defer body.Close()
-
-	bytes, err := ioutil.ReadAll(body)
+	input := validator.ValidationInput{}
+	err := json.NewDecoder(body).Decode(&input)
 	if err != nil {
-		gimlet.WriteJSONError(w, fmt.Sprintf("Error reading request body: %v", err))
+		gimlet.WriteJSONError(w, fmt.Sprintf("Error while unmarshalling JSON body: %v", err))
```

Comment:
Users are going to hit this if they're somehow still using the legacy structure referenced at 628. We should make this error more clear that they need to update their CLI to continue.

Reply:
I've actually gone ahead and decided against this idea - I've changed the way we unmarshal and remarshal the project config raw bytes in a way that will allow us to still just send one byte array. CLI updating won't be needed.